packages/@react-aria/color/src/useColorArea.ts

```diff
@@ -117,24 +106,26 @@ export function useColorArea(props: AriaColorAreaProps, state: ColorAreaState, i
       currentPosition.current = null;
       stateRef.current.setDragging(true);
     },
-    onMove({deltaX, deltaY, pointerType}) {
+    onMove({deltaX, deltaY, pointerType, shiftKey}) {
       if (currentPosition.current == null) {
         currentPosition.current = stateRef.current.getThumbPosition();
       }
       let {width, height} = containerRef.current.getBoundingClientRect();
       if (pointerType === 'keyboard') {
-        if (deltaX > 0 || deltaX < 0) {
-          stateRef.current[`${deltaX > 0 ? 'increment' : 'decrement'}X`]();
-        }
-        if (deltaY > 0 || deltaY < 0) {
-          stateRef.current[`${deltaY < 0 ? 'increment' : 'decrement'}Y`]();
+        if (deltaX > 0) {
+          stateRef.current.incrementX(shiftKey);
+        } else if (deltaX < 0) {
+          stateRef.current.decrementX(shiftKey);
+        } else if (deltaY > 0) {
+          stateRef.current.decrementY(shiftKey);
+        } else if (deltaY < 0) {
+          stateRef.current.incrementY(shiftKey);
         }
```

Comment:
So I take it this couldn't be unified with the mouse handling below?

Reply:
not without adjusting the math in setColorFromPoint, I haven't been able to figure it out yet, i think it would be good followup